File: Practice/src/coding_basic_training/SuffixArray.js
function solution(my_string) {
    var answer = [];
    for (let i = 0; i < my_string.length; i++) {
        answer.push(my_string.substring(i));
    }
    return answer.sort();
    //  from
    // return Array.from(my_string)
    //     .map((_, i) => my_string.substring(i))
    //     .sort();
}

File: Practice/src/coding_basic_training/SuffixArray.java
package coding_basic_training;

import java.util.ArrayList;
import java.util.Collections;

public class SuffixArray {
//	접미사 배열
	public String[] solution(String my_string) {
		ArrayList<String> li = new ArrayList<String>();
		for (int i = 0; i < my_string.length(); i++) {
			li.add(my_string.substring(i));
		}
		Collections.sort(li);
		return li.toArray(new String[li.size()]);
//		intstream
//        return IntStream.range(0, myString.length()).mapToObj(myString::substring).sorted().toArray(String[]::new);
	}
}